CMakeLists.txt

```diff
@@ -54,9 +43,9 @@ add_library(pass_through_filter_point_cloud2
 
 target_link_libraries(pass_through_filter_point_cloud2
     ${catkin_LIBRARIES}
-    #${PCL_LIBRARIES}
 )
 add_dependencies(pass_through_filter_point_cloud2 ${catkin_EXPORTED_TARGETS} ${${PROJECT_NAME}_EXPORTED_TARGETS})
+class_loader_hide_library_symbols(pass_through_filter_point_cloud2)
```

Comment:
This is a bit harder to explain, see https://github.com/ros-perception/perception_pcl/issues/9 . Basically, hiding the dynamic symbols helps interoperability of the plugin with other plugins. And it makes linking to the dynamic library impossible, which is however wanted.

Reply:
This one is great, I have never seen it :D